### PR TITLE
[mono][interp] Fix some leaks during compilation

### PR DESCRIPTION
**Pull request by BrzVlad**
Should fix out of memory on `System.Numerics.Tensors.Tests`. These seem to be sporadically failing, maybe since https://github.com/dotnet/runtime/pull/94555

### CODE REVIEW

**Comment by BrzVlad** [comment]
https://github.com/dotnet/runtime/issues/97124 https://github.com/dotnet/runtime/issues/96923

**Comment by srxqds** [comment]
> #97124
> 
> EDIT: #96923, this android failure doesn't seem to be interpreter

so dotnet8.0 have this problem?

File: src/mono/mono/mini/interp/transform.c [hunk 1]
```diff
@@ -3180,6 +3181,7 @@ interp_inline_newobj (TransformData *td, MonoMethod *target_method, MonoMethodSi
 	push_var (td, dreg);
 	return TRUE;
 fail:
+	mono_metadata_free_mh (mheader);
```
Comment (vargaz) [inline src/mono/mono/mini/interp/transform.c:3184]: Shouldn't this be if (mheader) ...

Comment (BrzVlad) [reply src/mono/mono/mini/interp/transform.c:3184]: could be, there is an explicit null check in `mono_metadata_free_mh` though.

**Review by vargaz — APPROVED** [review]

**Comment by vargaz** [comment]
The browser-wasm linux Release LibraryTests_EAT failure is unrealted.

**Pull request merged**